gen/templates/model.go

```diff
@@ -163,17 +163,22 @@ func (data {{camelCase .Name}}) toBody(ctx context.Context, state {{camelCase .N
 		data.{{toGoName .TfName}}.ElementsAs(ctx, &values, false)
 		body, _ = sjson.Set(body, "{{range .DataPath}}{{.}}.{{end}}{{.ModelName}}", values)
 	}
-	{{- else if isNestedListSet .}}
+	{{- else if isNestedListMapSet .}}
 	if len(data.{{toGoName .TfName}}) > 0 {
 		body, _ = sjson.Set(body, "{{range .DataPath}}{{.}}.{{end}}{{.ModelName}}", []interface{}{})
+		{{- if isNestedMap .}}
+		for key, item := range data.{{toGoName .TfName}} {
+			itemBody, _ := sjson.Set("{}", "name", key)
```

Comment:
Why `"name"`? What is the JSON we want to construct here?

Reply:
It might be more illustrative when I submit the functionality PR. The `Map` needed for any bulk-create functionality will always marshal to API bulk array:

```json
[
  {"name":"...","id":"...","attr":"..."},
  {"name":"...","id":"...","attr":"..."}
]
```

Reply:
Ok. It wasn't clear to me that this is specifically to support bulk APIs.